### PR TITLE
[MAINTENANCE] replace `isort` with `ruff` sorting rules

### PR DESCRIPTION
**Pull request by Kilo59**
### Changes proposed in this pull request:

- [x] - remove `isort` dependencies
- [x] - remove `isort` pre-commit hook
- [x] - update documentation
- [ ] - ~ruff pre-commit hook should run on full code-base (at least all of `great_expecatations/`)~
  - [ ] - ~for this we might want to wait until all of `great_expectations` is being linted (lint `great_expecations/dataset`)~
- ~update `expectation_diagnostics.py` to use `ruff` sorting~ - [see comment](https://github.com/great-expectations/great_expectations/pull/6907#discussion_r1084775270)
- [x] remove expectation_diagnostics `_check_linting` method
- [x] - `contrib/cli/great_expectations_contrib/commands.py` to use `ruff` sorting

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have made corresponding changes to the documentation

Follow-up to https://github.com/great-expectations/great_expectations/pull/6888

### CODE REVIEW

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py [hunk 1]
```diff
@@ -542,6 +542,7 @@ def _check_linting(expectation_instance) -> ExpectationDiagnosticCheckMessage:
                 )
             else:
                 black_ok = True
+            # TODO: update this for ruff (or don't)
             isort_ok = isort.check_code(
```
Comment (Kilo59) [inline great_expectations/core/expectation_diagnostics/expectation_diagnostics.py]: If we want to replace `isort` we need to replace this `expectation_diagnostics.py` check as well.
Either by running calling a function like with `isort` (don't know if this is possible) or wrapping it with a `subprocess.popen` call.

File: contrib/cli/great_expectations_contrib/commands.py [hunk 2]
```diff
@@ -61,6 +61,7 @@ def perform_check(suppress_output: bool) -> bool:
             "black --check .",
             "Please ensure that your files are linted properly with `black .`",
         ),
+        # TODO: update this (or don't)
         Command(
             "isort",
             "isort --profile black --check .",
```
Comment (Kilo59) [inline contrib/cli/great_expectations_contrib/commands.py]:
```suggestion
        Command(
            "ruff",
            "ruff --select I --fix .",
```

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py (hunk 1 above)
Comment (Kilo59) [reply great_expectations/core/expectation_diagnostics/expectation_diagnostics.py]: It was decided in our architecture review meeting that we should just remove this whole method and rely on our normal linting/formatting tools to enforce this.

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py [hunk 3]
```diff
@@ -454,115 +436,6 @@ def _check_renderer_methods(
...
-                    }
-                )
-
-        passed = black_ok and isort_ok and file_and_class_names_ok
-        return ExpectationDiagnosticCheckMessage(
-            message=message,
-            passed=passed,
-            sub_messages=sub_messages,
-        )
-
```
Comment (austiezr) [inline great_expectations/core/expectation_diagnostics/expectation_diagnostics.py:565]: This breaks the diagnostics workflow, but fix is simple. Call to `ExpectationDiagnostics._check_linting(...)` needs to be removed from `great_expectations.expectations.expectation.py:2209`

**Review by austiezr — DISMISSED** [review]
One note re: `ExpectationDiagnostics`; additionally, worth raising that the linting check will have to be removed from all references in the docs to the diagnostic checklist & our maturity levels, as it will no longer be performed as part of this loop. Happy take on that piece once this is merged.

File: contrib/cli/great_expectations_contrib/commands.py [hunk 4]
```diff
@@ -61,9 +61,10 @@ def perform_check(suppress_output: bool) -> bool:
             "black --check .",
             "Please ensure that your files are linted properly with `black .`",
         ),
+        # TODO: update this (or don't)
         Command(
-            "isort",
-            "isort --profile black --check .",
+            "ruff",
+            "ruff --select I --fix .",
             "Please ensure that your imports are sorted properly with `isort --profile black .`",
```
Comment (Kilo59) [inline contrib/cli/great_expectations_contrib/commands.py]:
```suggestion
            "Please ensure that your imports are sorted properly with `ruff --select I --fix .`",
```

File: great_expectations/core/expectation_diagnostics/expectation_diagnostics.py (hunk 3 above)
Comment (Kilo59) [reply great_expectations/core/expectation_diagnostics/expectation_diagnostics.py:565]: Done, thanks

**Comment by Kilo59** [comment]
Note this PR will fix the current build gallery failure.
Failure is because of a dependency conflict with `isort`.
https://dev.azure.com/great-expectations/great_expectations/_build/results?buildId=189123&view=logs&j=3a0a464f-5331-5fb4-9d39-5e0baff47fb3&t=6912365f-8191-5790-dbab-f01f88fbd16a

**Review by cdkini — APPROVED** [review]

**Comment by Kilo59** [comment]
After seeing all the test failures from `ExpectationsDiagnostic`...
I think it might actually be simpler to keep `_check_linting()` but replace it with a subprocess call to `ruff` 🤔

**Comment by austiezr** [comment]
How should we move forward here? From what I can see, there are 3 explicit failures linked to this change, all just needing output from `_check_linting` removed -- I don't have a particularly strong feeling either way re: keeping the linting check or not.

**Comment by Kilo59** [comment]
@austiezr I updated the 3 test failures to stop expecting `_check_linting`.
Merging.

**Pull request merged**